src/commands/git_secret_add.sh

```diff
@@ -32,9 +32,14 @@ function add {
     normalized_path=$(_git_normalize_filename "$item")
     path=$(_append_root_path "$normalized_path")
 
+    # check that the file is not tracked
+    if [[ $(git ls-files --error-unmatch "$item") ]]; then
```

Comment:
can we move this to a separate function?

Reply:
done!

Reply:
@joshrabinowitz thank you for doing this.
@sobolevn my apologies, didn't manage to find time.

I've a comment/question, the option **--error-unmatch** would display an error message for untracked files, like following:
```
error: pathspec 'untracked_file' did not match any file(s) known to git.
Did you forget to 'git add'?
```
Don't you think that the option **--cached** would be more appropriate ?
```
if [[ $(git ls-files --cached "$item") ]]; then
...
```

Reply:
@aymens  

That's not the error that happens in that case. 

If the file is completely new they new get the error:

````
% git secret add new_file_not_in_git.txt
git-secret: abort: these files are not in .gitignore: new_file_not_in_git.txt
````

If the new file is `git add`ed they get:
````
    % git secret add file_added_to_git.txt
    git-secret: abort: file tracked in git, consider using 'git rm --cached file_added_to_git.txt'
````

Reply:
and thanks for chiming in @aymens

Reply:
That's fine then!